core/src/main/scala/com/evolutiongaming/catshelper/LogOf.scala

```diff
@@ -8,6 +8,110 @@ import org.slf4j.{ILoggerFactory, LoggerFactory}
 
 import scala.reflect.ClassTag
 
+/** Factory of [[Log]] instances.
+  * 
+  * There are, currently, two ways of using the class and it is recommended to
+  * select one, describe it in project coding guidelines, and stick with it
+  * to avoid a confusion for the newcomers. Both ways are described below.
+  * 
+  * =Traditional approach=
+  * 
+  * The intented usage is to have a single instance of `LogOf` in the
+  * application and use it to create `Log` instances for each class.
+  *
+  * The following could be written somewhere in the application initialization
+  * code such as [[cats.effect.IOApp]] instance:
+  * {{{
+  * implicit val logOf = LogOf.slf4j[F]
+  * }}}
+  * 
+  * Then a typical example would be:
+  * {{{
+  * class UserService[F[_]: Monad](log: Log[F]) {
+  * 
+  *  def create(user: User): F[Unit] =
+  *    for {
+  *      _ <- log.info(s"Creating user...")
+  *      _ <- ...
+  *    } yield ()
+  *   
+  * }
+  * 
+  * object UserService {
+  * 
+  *   def of[F[_]: LogOf]: F[UserService[F]] =
+  *     for {
+  *       log <- LogOf[F].forClass[UserService]
+  *       service = new UserService[F](log)
+  *     } yield service
+  * 
+  * }
```

Comment:
Consider swapping the order of `class` and `object` here. When reading from top to bottom it would make it easier to trace the sequence "request `LogOf`" > "use it to create `Log`" > "use `Log`".

```suggestion
  * object UserService {
  * 
  *   def of[F[_]: LogOf]: F[UserService[F]] =
  *     for {
  *       log <- LogOf[F].forClass[UserService]
  *       service = new UserService[F](log)
  *     } yield service
  * 
  * }
  * 
  * class UserService[F[_]: Monad](log: Log[F]) {
  * 
  *  def create(user: User): F[Unit] =
  *    for {
  *      _ <- log.info(s"Creating user...")
  *      _ <- ...
  *    } yield ()
  *   
  * }
```